src/fides/api/ops/schemas/dataset.py

```diff
@@ -15,17 +15,21 @@
 )
 
 
-def validate_data_categories_against_db(dataset: Dataset, db: Session) -> None:
+def get_data_categories_from_db(db: Session) -> List[FidesKey]:
+    """Query for existing data categories in the db using a synchronous session"""
+    return [cat[0] for cat in db.query(DataCategory.fides_key).all()]
```

Comment:
Refactored so I could share the `validate_data_categories_against_db` method on the ops side by passing in a list of data categories since one endpoint has access to an async session and the other has access to a sync session right now.